docs/ci-cd.md

```diff
@@ -12,3 +12,43 @@ The following secrets should be kept in the repository:
 | BOT_TOKEN | PAT (Personal Access Token) of rh-operator-bundle-test-e2e | E2E tests- fake "partner" account, allows access to repository operator-pipelines-test |
 | KUBECONFIG | Kubeconfig of the cluster, where E2E tests should run | E2E tests- resources are deployed to the namespace $SHORT_COMMIT_HASH in this cluster |
 | VAULT_PASSWORD | Password to the Ansible Vault stored in the repository | E2E tests- password is needed to decrypt the secrets and thus- recreate the resources |
+
+## E2E tests
+
+### When they run?
+
+E2E tests are stage of the CI/CD that runs only in two cases:
+- On merge to main and before deployment
+- On desire (manual action- by clicking "run workflow" via GitHub UI)
+
+There are future plans to run them in cronjob as a part of the QA workflow.
+
+### E2E tests stages?
+
+- prepare the environment- script [ansible/init-custom-env.sh](../ansible/init-custom-env.sh):
+  - prepare the custom project in OpenShift preprod cluster
+  - install all of the pipelines dependencies in this project
+- run the pipelines with `tkn`
+- check if pipelines passed- using `oc`
+- remove the project, and thus- clean up all associated resources
+
+To read more about the `init-custom-env`, see [developer-guide.md](developer-guide.md).
+
+### E2E tests dependencies
```

Comment:
Is that enough information?

Reply:
Can you also describe how are the individual pipelines triggered and if there are any data dependencies between pipelines?